src/core/methods/createChannel.ts

```diff
@@ -3,134 +3,74 @@
 import {
   ChannelState,
   FeedType,
-  MembraneType,
-  FluxExpressionReference,
 } from "@/store/types";
-import { v4 } from "uuid";
-import { Perspective, Link } from "@perspect3vism/ad4m";
 import type { PerspectiveHandle } from "@perspect3vism/ad4m";
-import { createNeighbourhoodMeta } from "./createNeighbourhoodMeta";
-import { SOCIAL_CONTEXT_OFFICIAL } from "@/constants/languages";
 import { ad4mClient } from "@/app";
 import { CHANNEL } from "@/constants/neighbourhoodMeta";
+import { useDataStore } from "@/store/data";
+import { useAppStore } from "@/store/app";
 
 interface ChannelProps {
   channelName: string;
   creatorDid: string;
   sourcePerspective: PerspectiveHandle;
-  membraneType: MembraneType;
-  typedExpressionLanguages: FluxExpressionReference[];
 }
 
 export async function createChannel({
   channelName,
   creatorDid,
   sourcePerspective,
-  membraneType,
-  typedExpressionLanguages,
 }: ChannelProps): Promise<ChannelState> {
-  const perspective = await ad4mClient.perspective.add(channelName);
-  console.debug("Created new perspective with result", perspective);
-  const socialContextLanguage =
-    await ad4mClient.languages.applyTemplateAndPublish(
-      SOCIAL_CONTEXT_OFFICIAL,
-      JSON.stringify({
-        uid: v4().toString(),
-        name: `${channelName}-social-context`,
-      })
-    );
-  console.debug(
-    "Created new social context language wuth result",
-    socialContextLanguage
-  );
-
-  //Publish perspective
-  const metaLinks = await createNeighbourhoodMeta(
-    channelName,
-    "",
-    creatorDid,
-    typedExpressionLanguages
-  );
-
-  const meta = new Perspective(metaLinks);
+  const dataStore = useDataStore();
+  const appStore = useAppStore();
 
-  const neighbourhood = await ad4mClient.neighbourhood.publishFromPerspective(
-    perspective.uuid,
-    socialContextLanguage.address,
-    meta
-  );
-  console.debug("Create a neighbourhood with result", neighbourhood);
+  const channel = dataStore.channels[channelName];
+  if (!channel || channel.sourcePerspective !== sourcePerspective.uuid) {
+    const addLinkToChannel = await ad4mClient.perspective.addLink(
+      sourcePerspective.uuid,
+      {
+        source: sourcePerspective.sharedUrl!,
```

Comment:
This is this' perspective's sharedUrl? I think doing this is (in general) a bit problematic, which is why we use `ad4m://self` in the hackmd to refer to the perspective/neigbourhood we're in.

Reply:
I quite like having the sharedUrl as the source since it represents the object that is the neighbourhood/perspective. ad4m://self almost seems like it could represent the ad4m ontology itself. In any case I am not attached to either outcome.